src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -1618,6 +1618,21 @@ async def hook_license_update(middleware, *args, **kwargs):
     await middleware.call('service.restart', 'failover')
     await middleware.call('failover.status_refresh')
 
+async def hook_post_rollback_setup_ha(middleware, *args, **kwargs):
+    """
+    This hook needs to be run after a NIC rollback operation is run
+    on a TrueNAS HA system
+    """
+    if not await middleware.call('failover.licensed'):
+        return
+
+    if await middleware.call('failover.status') == 'UNKNOWN':
```

Comment:
I am not sure we can trust failover.status to tell us that, how about check NO_PING in disabled_reasons or actually try to ping it before? Or even catch an specific exception?

I imagine either would these would be more bullet proof